crates/store/src/db/models/queries/insertions.rs

```diff
@@ -285,6 +295,50 @@ fn build_insert_delta(account: &Account) -> Result<AccountDelta, DatabaseError>
     Ok(AccountDelta::new(account.id(), storage_delta, vault_delta, account.nonce())?)
 }
 
+/// Insert an account storage map value into the DB using the given [`SqliteConnection`].
+///
+/// This function will set `is_latest_update=true` for the new row and update any existing
+/// row with the same `(account_id, slot, key)` tuple to `is_latest_update=false`.
+///
+/// # Returns
+///
+/// The number of affected rows.
+pub(crate) fn insert_account_storage_map_value(
+    conn: &mut SqliteConnection,
+    account_id: AccountId,
+    block_num: BlockNumber,
+    slot: u8,
+    key: Word,
+    value: Word,
+) -> Result<usize, DatabaseError> {
+    // First, update any existing rows with the same (account_id, slot, key) to set
+    // is_latest_update=false
+    let update_count = diesel::update(schema::account_storage_map_values::table)
+        .filter(
+            schema::account_storage_map_values::account_id
+                .eq(account_id.to_bytes())
+                .and(schema::account_storage_map_values::slot.eq(slot_to_raw_sql(slot)))
+                .and(schema::account_storage_map_values::key.eq(&key.to_bytes()))
+                .and(schema::account_storage_map_values::is_latest_update.eq(true)),
+        )
+        .set(schema::account_storage_map_values::is_latest_update.eq(false))
+        .execute(conn)?;
+
+    // Insert the new row with is_latest_update=true
+    let insert_count = diesel::insert_into(schema::account_storage_map_values::table)
+        .values(&[(
+            schema::account_storage_map_values::account_id.eq(account_id.to_bytes()),
+            schema::account_storage_map_values::block_num.eq(block_num.to_raw_sql()),
+            schema::account_storage_map_values::slot.eq(slot_to_raw_sql(slot)),
+            schema::account_storage_map_values::key.eq(key.to_bytes()),
+            schema::account_storage_map_values::value.eq(value.to_bytes()),
+            schema::account_storage_map_values::is_latest_update.eq(true),
+        )])
+        .execute(conn)?;
```

Comment:
Not for this PR: I think this works fine because we are working with an in-memory database, but if we were working with something like Postgres running on a separate machine, we'd probably want to send both queries together to the server. And probably send a batch of updates to minimize the number of network requests.

Reply:
I meant to add the change here, but I had modified this to use a single transaction in the follow up: https://github.com/0xMiden/miden-node/pull/1140/files#diff-1a7b5569fe361ba6496af6cef83594571e7e29590f0ced96fbeadfaf51298146R320